subprojects/build-operations/src/main/java/org/gradle/internal/operations/BuildOperationExecutionListener.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.operations;
+
+import javax.annotation.Nullable;
+
+interface BuildOperationExecutionListener {
```

Comment:
❓ How does this relate to `BuildOperationListener`?

Reply:
It is an internal thing to the `DefaultBuildOperationRunner`. I guess it makes sense to move it back to an inner-type.